### PR TITLE
Refactoring of deployment tools

### PR DESCRIPTION
**Pull request by druzhkov-paul**
Summary of introduced changes:

- [x] All deployment-related routines were moved to a dedicated subpackage: `mmdet.utils.deployment`
- [x] `test_exported.py` script has been unified to share as much logic between ONNXRuntime and OpenVINO branches as possible
- [x] Added a flag to disable tracer stubs during export to obtain ONNXRuntime-compatible models
- [x] Added stubs for pyramid ROIAlign feature extractor

### CODE REVIEW

**Comment by druzhkov-paul** [comment]
@Ilya-Krylov please take a look

File: docs/GETTING_STARTED.md [hunk 1]
```diff
@@ -185,13 +185,13 @@ To further convert ONNX model to IR you can run the script:
 In order to test the model being exported via a `tools/export.py` script run the following command:
 
 ```shell
-# python tools/test_exported.py config.py ${DEPLOY_DIR}/checkpoint.xml --out /tmp/out.pkl --backend openvino --do_not_normalize
```
Comment (Ilya-Krylov) [inline docs/GETTING_STARTED.md:188]: please fix it in other *.md files

Comment (druzhkov-paul) [reply docs/GETTING_STARTED.md:188]: Which ones?

File: mmdet/utils/deployment/openvino_backend.py [hunk 2]
```diff
@@ -157,27 +157,28 @@ def configure_outputs(self, net, required_outputs):
...
                 else:
-                    raise ValueError('Failed to identify output {}'.format(required_output))
+                    raise ValueError('Failed to identify output "{}"'.format(required_output))
         return net_outputs_mapping
 
     def rename_outputs(self, outputs):
         return {self.net_outputs_mapping[k]: v for k, v in outputs.items() if k in self.net_outputs_mapping}
 
-    def forward(self, inputs):
+    def normalize_inputs(self, inputs):
```
Comment (Ilya-Krylov) [inline mmdet/utils/deployment/openvino_backend.py]: is it a good name? maybe `remap_inputs` or `fill_inputs` or something? but not normalize

File: tools/convert_to_ir.py [hunk 3]
```diff
@@ -14,16 +14,21 @@
 # See the License for the specific language governing permissions
 # and limitations under the License.
 
-import argparse
+import os
```
Comment (Ilya-Krylov) [inline tools/convert_to_ir.py:17]: why did you sort it this way?

Comment (druzhkov-paul) [reply tools/convert_to_ir.py:17]: 1. Standard library imports.
1. Related third party imports.
1. Local application/library specific imports.

File: docs/GETTING_STARTED.md (hunk 1 above)
Comment (Ilya-Krylov) [reply docs/GETTING_STARTED.md:188]: in OTE, non-existing PR that you should create :)

File: mmdet/utils/deployment/onnxruntime_backend.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                    helper.make_empty_tensor_value_info(output_id))
+            else:
+                extra_outputs.append(value_info)
+
+        self.model.graph.output.extend(extra_outputs)
+        self.output_names.extend(output_ids)
+        self.session = onnxruntime.InferenceSession(
+            self.model.SerializeToString(), self.sess_options)
+
+    def normalize_inputs(self, inputs):
```
Comment (Ilya-Krylov) [inline mmdet/utils/deployment/onnxruntime_backend.py]: ditto

**Comment by Ilya-Krylov** [comment]
good job

**Review by Ilya-Krylov — APPROVED** [review]

File: mmdet/utils/deployment/openvino_backend.py (hunk 2 above)
Comment (druzhkov-paul) [reply mmdet/utils/deployment/openvino_backend.py]: Renamed to `unify_inputs`.

File: docs/GETTING_STARTED.md (hunk 1 above)
Comment (druzhkov-paul) [reply docs/GETTING_STARTED.md:188]: Ok then :smile:

**Pull request merged**